source/session.ts

```diff
@@ -316,15 +316,15 @@ export class Session {
         // to timezone naive string.
         return {
           __type__: "datetime",
-          value: data.utc().format(ENCODE_DATETIME_FORMAT),
+          value: date,
         };
       }
 
       // Ensure that the moment object is in local time zone and format
       // to timezone naive string.
       return {
         __type__: "datetime",
-        value: data.local().format(ENCODE_DATETIME_FORMAT),
+        value: moment(date).local().format(ENCODE_DATETIME_FORMAT),
```

Comment:
Should we target the `v2` for these changes as well? I am not really sure how to test this for regressions, so feels safer to do this while also removing this branch.

Reply:
i have intentionally separated code for this PR to be backwards compatible by making sure all moment objects are passed in the same manner as before, so I hope we can target this for a minor release. we can add additional tests for this if there are certain use cases you are unsure about.

Reply:
I want to ensure that this path hasn't changed, without having to disable timezone support locally.

Can we add a test for:

```
const date = moment()
const convertedDate = convertToISOString(date)
const encodedDate = moment(date).local().format(ENCODE_DATETIME_FORMAT)

assert date.local().format(ENCODE_DATETIME_FORMAT) === encodedDate
```

Reply:
please see new tests!